inet/vibe/inet/url.d

```diff
@@ -19,11 +19,75 @@ import std.exception;
 import std.string;
 import std.traits : isInstanceOf;
 import std.ascii : isAlpha, isASCII, toLower;
-import std.uri: encode;
+import std.uri: decode, encode;
 
 import core.checkedint : addu;
 
 
+/** Parses a user-entered URL with relaxed parsing rules.
+
+	Unlike `URL.parse`, this allows the URL to use special charaters as part of
+	the host name and path, automatically emplying punycode or percent-encoding
+	to convert this to a valid URL.
+
+	Also, relative URL references are supported by supplying a base URL.
```

Comment:
There's a few typos (`charaters`, `emplying`), but I'm a bit confused about what this is supposed to do.
Will ask questions in the review comment and then suggest a rewrite.

Reply:
I'll fix those, this was mostly copy-pasted from `parsePlain`.